timetabler/ssc/course.py

```diff
@@ -1,3 +1,6 @@
+import re
+
+
 class Course(object):
     def __init__(self, dept, number,
                  lectures=None, labs=None, tutorials=None, discussions=None):
```

Comment:
Okay, so there should be a `skip_duplicates=False` kwarg here. You will need to pass this all the way through, but it's fine (i.e., example -> Scheduler -> get_course -> Course). This should be controllable by the user at the top-level, like you said.